src/main/java/bio/terra/service/filedata/azure/AzureSynapsePdao.java

```diff
@@ -552,6 +555,75 @@ public void createSnapshotRowIdsParquetFile(
     executeSynapseQuery(sqlCreateRowIdTable.render());
   }
 
+  public Map<String, Long> createSnapshotParquetFilesByQuery(
+      AssetSpecification assetSpec,
+      UUID snapshotId,
+      String datasetDataSourceName,
+      String snapshotDataSourceName,
+      String translatedQuery,
+      Boolean isGlobalFieldIds)
+      throws SQLException {
+    Map<String, Long> tableRowCounts = new HashMap<>();
+
+    // First handle root table
+    AssetTable rootTable = assetSpec.getRootTable();
+    String rootTableName = rootTable.getTable().getName();
+
+    // Get columns to include for root table
+    List<SynapseColumn> columns =
+        rootTable.getColumns().stream()
+            .map(c -> c.getDatasetColumn())
+            .collect(Collectors.toList())
+            .stream()
+            .map(Column::toSynapseColumn)
+            .collect(Collectors.toList());
```

Comment:
Actually, zooming out a bit has made me realize that we can eliminate intermediate conversions to / from streams.  This could be one way:

```
    List<SynapseColumn> columns =
        rootTable.getColumns().stream()
            .map(c -> Column.toSynapseColumn(c.getDatasetColumn()))
            .toList();
```

Reply:
Good point! Taking another look at this, it turns out I already had a helper method on AssetTable to get the synapse columns. Let me know what you think of [these changes](https://github.com/DataBiosphere/jade-data-repo/pull/1399/commits/671efea339cc0463bf90b7ccf39bac51cd9fc4de).

Reply:
Nice! Even as I made this suggestion, I thought the code looked familiar from a prior PR, I remember Phil made a similar suggestion on stream consolidation.